src/xdocs/checks/coding/illegaltokentext.xml

```diff
@@ -52,26 +51,28 @@
               <td>tokens to check</td>
               <td>subset of tokens
                 <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#NUM_DOUBLE">
-                NUM_DOUBLE</a>
+                    NUM_DOUBLE</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#NUM_FLOAT">
-                NUM_FLOAT</a>
+                    NUM_FLOAT</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#NUM_INT">
-                NUM_INT</a>
+                    NUM_INT</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#NUM_LONG">
-                NUM_LONG</a>
+                    NUM_LONG</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#IDENT">
-                IDENT</a>
+                    IDENT</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#COMMENT_CONTENT">
-                COMMENT_CONTENT</a>
+                    COMMENT_CONTENT</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#STRING_LITERAL">
-                STRING_LITERAL</a>
+                    STRING_LITERAL</a>
                 , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CHAR_LITERAL">
-                CHAR_LITERAL</a>
-                ,<a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#TEXT_BLOCK_CONTENT">
-                TEXT_BLOCK_CONTENT</a>
-                .
+                    CHAR_LITERAL</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#TEXT_BLOCK_CONTENT">
+                    TEXT_BLOCK_CONTENT</a>
+                  .
+              </td>
+              <td>
+                  .
               </td>
-              <td><code>empty</code></td>
```

Comment:
I am not expecting any changes here unless https://github.com/checkstyle/checkstyle/issues/13876 or some other issue.

Reply:
@rnveach, i have been spending some time with the code. And, i have slightly better understanding of the issue. I have a question. In the code, we are comparing defaultTokenText and expectedDefaultTokenText. defaultTokenText makes a call to DOM API. And, surprisingly, it return "." for an empty token. And, the expectedDefaultTokenText makes a call to the method fo CheckUtil returning empty. And, as seen, there is assertion failure. 

```
Empty tokens should have 'empty' string in xdoc
expected: empty
but was : .
Expected :empty
Actual   :.
```

One solution would be to provide an implementation of getTokenText of DOM API. Haven't probed much on that aspect. Or, adjust the code to make sure we match up to DOM. However, the issue is not limited to this. I see it coming in many macro related migration. What according to you would be an apt solution? And, as you had mentioned, i feel this isn't a standaridisation issue. But, the maybe the way the javadocs is being processed or written gives us some other result than expected. 
<img width="1094" alt="image" src="https://github.com/checkstyle/checkstyle/assets/95259545/17a43282-8333-4289-8fa8-5ae0efc403c7">

Reply:
done